### PR TITLE
precommit: introduce Black & docs

### PR DESCRIPTION
**Pull request by Borda**
This is about eventually accepting Black formatting, which I know could be very opinionated, but at least take it as a suggestion to show how the codebase could look... In fact, using Ruff formatting, which claims compatibility with Black, see https://docs.astral.sh/ruff/formatter/#black-compatibility

### CODE REVIEW

**Comment by Borda** [comment]
There copuld be some cases when the formating is not your Cup of tea, but over all, I find the codebase more readable... :)

**Comment by Borda** [comment]
> I can do with everything but the long lines.

it is one of the parameters, so what is your preferred line length?

**Comment by shaypal5** [comment]
Pep8's 79 characters.
@Borda

**Comment by shaypal5** [comment]
Nice. This looks done. Are done? Should I review? @Borda

**Review by shaypal5 — APPROVED** [review]

File: tests/test_general.py [hunk 1]
```diff
@@ -86,14 +85,8 @@ def _calls_wait_for_calc_timeout_fast(res_queue):
...
     res_queue = queue.Queue()
-    thread1 = threading.Thread(
-        target=_calls_wait_for_calc_timeout_fast,
-        kwargs={'res_queue': res_queue},
-        daemon=True)
-    thread2 = threading.Thread(
-        target=_calls_wait_for_calc_timeout_fast,
-        kwargs={'res_queue': res_queue},
-        daemon=True)
+    thread1 = threading.Thread(target=_calls_wait_for_calc_timeout_fast, kwargs={"res_queue": res_queue}, daemon=True)
```
[the remark this answers is not in the document]
Comment (Borda) [reply tests/test_general.py]: true, not sure why

[the remark this answers is not in the document]
Comment (shaypal5) [reply tests/test_general.py]: No no. It's outdated.

**Pull request merged**